### ISSUE TITLE
win_regmerge: Not idempotent, no check-mode and no diff support

### ISSUE THREAD

**Issue by dagwieers** [ansible/ansible#18819]
_From @dagwieers on May 28, 2016 6:44_

##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

win_regmerge
##### ANSIBLE VERSION

v2.1
##### SUMMARY

The win_regmerge module does not check each registry item it modifies to validate if there was a change. It instead always applies the list of changes regardless.

The proper implementation of win_regmerge would read the file and verify each individual setting with the registry, and report back changes and diff-output.

_Copied from original issue: ansible/ansible-modules-extras#2332_

**Comment by dagwieers** [comment]
_From @jhawkesworth on June 6, 2016 22:23_

Hi sorry for delay replying, illness and other things have been in the way.
My intention was to have a way of bulk updating the registry - win_regedit is best for individual registry changes, but can be slow to apply lots of changes.  I will look again and see if it can be improved though.

**Comment by dagwieers** [comment]
@jhawkesworth I am considering parsing .reg files in python and verifying existence myself. This would allow for check-mode _and_ diff-mode, but requires creating, comparing and deleting entries.

**Comment by dagwieers** [comment]
_From @jhawkesworth on June 11, 2016 20:11_

@dagwieers - feel free to take a stab at this - you might find there's some useful logic in win_regedit.  Its kind of a tricky one as there isn't really an equivalent of the registry on linux, so at some point you wind up dealing with a conversion between the file and the registry.
Right now this module is just slightly better than using raw to run reg.exe in that it won't report changed every time it is run.

**Comment by dagwieers** [comment]
_From @ansibot on August 4, 2016 18:58_

@jhawkesworth, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
Still interested in this, but it is a personal project to improve Windows modules and I don't have any personal time ATM.

waiting_for_contributor OR waiting_for_free_time :-)

**Comment by dagwieers** [comment]
_From @jhawkesworth on August 15, 2016 10:39_

Right now its not a priority for me to make any improvements to this module, (although it could definitely be improved as you describe, I am not going to get to it any time soon).  My original intention was to a/ not use 'raw' to merge registry files and b/ have a quick way of merging multiple registry entries in a single module call - win_regedit is definitely the way to go for single registry changes, and would probably be the easiest place to implement diff and check mode (which is lacking for almost all win_ modules right now).

What to do with this issue then?

To me its a feature request rather than a bug report.  If you agree @dagwieers  lets turn it into a feature request / proposal so it can go on the backlog.

**Comment by dagwieers** [comment]
Sure.

waiting_for_contributor

**Comment by dagwieers** [comment]
_From @ansibot on September 11, 2016 15:1_

@jhawkesworth, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @jhawkesworth on September 12, 2016 16:34_

Changing to a feature idea as agreed above due to ENOTIME from anyone right now to improve this.

**Comment by dagwieers** [comment]
_From @ansibot on October 6, 2016 17:56_

@jhawkesworth, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @jhawkesworth on October 6, 2016 19:32_

Oops, I hadn't got my head round the bot commands when I last touched this.

needs_contributor

**Comment by dagwieers** [comment]
_From @ansibot on December 6, 2016 21:58_

This repository has been locked. All new issues and pullrequests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pullrequest to the ansible/ansible repo.

**Comment by ansibot** [comment]
cc @jborean93
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @nitzmahone
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @if-meaton
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @ShachafGoldstein
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jborean93** [comment]
While this could be implemented this is not something we are looking to do at this moment. If someone wishes to tackle this they can raise a PR on https://github.com/ansible-collections/community.windows which is where the module is not located.
